spec/policies/project_policy_spec.rb

```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+describe ProjectPolicy do
+  subject { described_class }
+
+  permissions :update? do
+    let(:user) { FactoryBot.build(:user) }
+    let(:locked_project) { FactoryBot.create(:project, locked_at: Time.current) }
+    let(:project) { FactoryBot.create(:project, locked_at: nil) }
+
+    context "when project doesn't belong to a parent project" do
+      it "denies access if project is locked" do
+        expect(subject).not_to permit(user, locked_project)
+      end
+
+      it "grants access if project is not locked" do
+        expect(subject).to permit(user, project)
+      end
+    end
+
+    context "when sub project belongs to a locked parent" do
+      let(:sub_project) { FactoryBot.create(:project, locked_at: nil, parent: locked_project) }
+
+      it "denies access" do
+        expect(subject).not_to permit(user, sub_project)
+      end
+    end
+
+    context "when sub project belongs to an unlocked parent" do
+      let(:sub_project) { FactoryBot.create(:project, locked_at: nil, parent: project) }
```

Comment:
And here `build` or `build_stubbed`

Reply:
> Great job so far!! tada
> 
> I think the last thing I just picked up is that you can still bulk delete stories on a locked project. ![image](https://user-images.githubusercontent.com/3785596/175978693-308371fa-824f-4505-82c2-505b1044eb25.png)
> 
> ![image](https://user-images.githubusercontent.com/3785596/175978772-100b9415-ac02-45e8-815d-31a0b0fc34d8.png)

Again thank you for pointing this out, fixed